datafusion/src/logical_plan/plan.rs

```diff
@@ -141,7 +137,7 @@ pub enum LogicalPlan {
     /// Produces rows from a table provider by reference or from the context
     TableScan {
         /// The name of the table
-        table_name: String,
+        table_name: Option<String>,
```

Comment:
I don't fully understand how a table scan (which results in a relation) can end up without a name. What is an example of when we need to use a `None` table_name?

Reply:
This is needed because all the non-table scan all eventually goes into TableScan, for example: https://github.com/apache/arrow-datafusion/blob/245f0b8a68c5763a236aef3e727f0502188d0bfa/datafusion/src/logical_plan/builder.rs#L110-L117. Currently we are using empty str `''` to imply scan without table names, which makes the relation name match a little bit less readable downstream in the code base.

Reply:
I guess I don't fully understand how we could be scanning relations that don't have names (like how do you refer to them in SQL?)

Reply:
In SQL this would never happen. It's more of a thing in spark land where for simple queries, people usually just load a csv or parquet partition into a dataframe without any table registration. So for example, for dataframe users, they could start with a `Context.read_csv` call and only reference columns by unqualified names in subsequent transformations.

Reply:
I see -- thank you for the explanation. In this case in DataFusion, we could perhaps register the data with either a temporary name (or the csv file name or something).

Reply:
Updated code to assign `?table?` as name for unnamed tables. TableScan name is not an Option anymore.